pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,394 @@
+            self.adapter.auto = 1
+            self.adapter.connection.query_delay = 0
+            self.adapter.gpib_read_timeout = 500
+            self.adapter.connection.timeout = 700
+            self.adapter.eoi = 1
+            self.adapter.eos = "\n"
+            self.adapter.connection.read_termination = self.adapter.eos
+            self.adapter.connection.read_termination = "\r\n"
+            self.adapter.write("++eot_enable 0")
+            self.adapter.flush_read_buffer()
+            self.write("FORM4")
+            self.adapter.flush_read_buffer()
+
+    @cached_property
+    def id(self):
+        """Get the identification of the instrument."""
+        if self.SCPI:
+            result = self.ask("*IDN?").strip("\n")
+            self._manu, self._model, _, self._fw = result.strip("\r\n").split(",")
+            return result
+        else:
+            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+
+    @cached_property
+    def manu(self):
+        """Get the manufacturer of the instrument."""
+        try:
+            return self._manu
+        except AttributeError:
+            self.id
+            return self._manu
+
+    @cached_property
+    def model(self):
+        """Get the model of the instrument."""
+        try:
+            return self._model
+        except AttributeError:
+            self.id
+            return self._model
+
+    @cached_property
+    def fw(self):
+        """Get the firmware of the instrument."""
+        try:
+            return self._fw
+        except AttributeError:
+            self.id
+            return self._fw
+
+    @cached_property
+    def sn(self):
+        """Get the serial number of the instrument"""
+        if self.SCPI:
+            result = self.ask("OUTPSERN").strip("\r\n")
+            return result
+        else:
+            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+
+    @cached_property
+    def options(self):
+        """Get the options of the instrument"""
+        if self.SCPI:
+            result = self.ask("OUTPOPTS").strip("\r\n").strip()
+            return result
+        else:
+            raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+
+    def set_fixed_frequency(self, frequency):
+        """Sets the scan to be of only one frequency in Hz"""
+        self.start_frequency = frequency
+        self.stop_frequency = frequency
+        self.scan_points = 3
+
+    @property
+    def parameter(self):
+        for parameter in HP8753E.SCATTERING_PARAMETERS:
+            if int(self.ask(f"{parameter}?")) == 1:
+                return parameter
+        return None
+
+    @parameter.setter
+    def parameter(self, value):
+        if value in HP8753E.SCATTERING_PARAMETERS:
+            self.write("%s" % value)
+        else:
+            raise Exception("Invalid scattering parameter requested" " for Hewlett Packard 8753E")
```

Comment:
Can't you do that with our property creators (`Instrument.control`)?

Reply:
I don't know how to rewrite this property to query the instrument for each possible measurement parameter and break when one of the properties finally responds with a '1'.

So if the instrument was set to "S22", I would need to ask it "S11?" which it will reply with "0". Same for "S12" and "S21". Once it gets asked "S22" and gets the '1' it should break. I wouldn't want to ask it "AR?" "BR?", "AB?", "MEASA?" etc for the other measurement parameters since that will just add time to complete the test.

It looks like `Instrument.control` doesn't have the hooks to iterate through asking the potentially 10 queries.

Reply:
Yes the general "Control" does not do that. Therefore, it is fine to use your custom property.